webapp/src/main/java/com/box/l10n/mojito/rest/scheduledjob/ScheduledJobWS.java

```
@@ -35,18 +39,159 @@ public class ScheduledJobWS {
+          null);
+    }
+    if (scheduledJob.getCron() == null || scheduledJob.getCron().isEmpty()) {
+      return createResponse(
+          HttpStatus.BAD_REQUEST,
+          ScheduledJobResponse.Status.FAILURE,
+          "Cron expression must be provided to create a job",
+          scheduledJob.getUuid());
+    }
+    try {
+      if (scheduledJob.getUuid() == null) {
+        scheduledJob.setUuid(UUID.randomUUID().toString());
+      }
+      if (scheduledJob.getJobStatus() == null) {
+        scheduledJob.setJobStatus(
+            scheduledJobStatusRepository.findByEnum(ScheduledJobStatus.SCHEDULED));
+      }
+      if (scheduledJob.getJobType() != null && scheduledJob.getJobType().getId() != null) {
+        scheduledJob.setJobType(
+            scheduledJobTypeRepository
+                .findById(scheduledJob.getJobType().getId())
+                .orElseThrow(
+                    () ->
+                        new ResponseStatusException(
+                            HttpStatus.NOT_FOUND,
+                            "Job type not found with id: " + scheduledJob.getJobType().getId())));
+      } else {
+        throw new ResponseStatusException(
+            HttpStatus.BAD_REQUEST, "Job type must be provided to create a job");
+      }
+
+      scheduledJobRepository.save(scheduledJob);
+      scheduledJobManager.scheduleJob(scheduledJob);
+
+      logger.info(
+          "Job '{}' for repository '{}' was created.",
+          scheduledJob.getUuid(),
+          scheduledJob.getRepository().getName());
+      return ResponseEntity.status(HttpStatus.OK)
+          .body(
+              new ScheduledJobResponse(
+                  ScheduledJobResponse.Status.SUCCESS, "Job created successfully", scheduledJob));
+    } catch (Exception e) {
+      logger.error("Error creating job", e);
+      return createResponse(
+          HttpStatus.BAD_REQUEST,
+          ScheduledJobResponse.Status.FAILURE,
+          "Error creating job: " + e.getMessage(),
```

I'm not sure if we should bubble up the full exception message that might contain sensitive version info. Rather give a vague error here as we can view the logs to get the details